internal/devices/reinstall.go

```diff
@@ -42,7 +42,7 @@ func (c *Client) Reinstall() *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 
 			request := packngo.DeviceReinstallFields{
-				OperatingSystem: "",
+				OperatingSystem: operatingSystem,
 				PreserveData:    preserveData,
 				DeprovisionFast: preserveData,
```

Comment:
Does it make sense to pin `DeprovisionFast` to `preserveData`?
What are the benefits of preserving data without deprovisioning fast?

Reply:
I had this reasoning in the wrong place... in a comment on the previous pr https://github.com/equinix/metal-cli/pull/233/files#r1019605111. I've changed the code so they aren't tied together. Its actually the other way around. `PreserverData` currently implies `DeprovisionFast` in the deprovision process, `DeprovisionFast` currently is only useful to avoid the slow wipe of disks that need it. In the future `DeprovisionFast` could also skip lengthy inventory collection steps that don't really need to happen for a reinstall.